x/autopilot/keeper/redeem_stake.go

```diff
@@ -18,74 +17,64 @@ import (
 func (k Keeper) TryRedeemStake(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
-	newData transfertypes.FungibleTokenPacketData,
-	packetMetadata types.StakeibcPacketMetadata,
+	transferPacketData transfertypes.FungibleTokenPacketData,
+	autopilotMetadata types.StakeibcPacketMetadata,
 ) error {
 	params := k.GetParams(ctx)
 	if !params.StakeibcActive {
 		return fmt.Errorf("packet forwarding param is not active")
 	}
 
-	// In this case, we can't process a liquid staking transaction, because we're dealing IBC tokens from other chains
-	if !transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), newData.Denom) {
-		return fmt.Errorf("the ibc tokens are not supported for redeem stake")
+	// At this point in the stack, the denom's in the packet data appear as they existed on the sender zone,
+	//   but as a denom trace instead of a hash
+	// Meaning, for native stTokens, the port and channel on the host zone are part of the denom
+	//   (e.g. transfer/{channel-on-hub}/stuatom)
+	// Only stride native stTokens can be redeemed, so we confirm that the denom's prefix matches
+	//   the packet's "source" channel (i.e. the channel on the host zone)
+	if !transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), transferPacketData.Denom) {
+		return fmt.Errorf("the ibc token %s is not supported for redeem stake", transferPacketData.Denom)
 	}
 
 	voucherPrefix := transfertypes.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
-	stAssetDenom := newData.Denom[len(voucherPrefix):]
+	stAssetDenom := transferPacketData.Denom[len(voucherPrefix):]
 	if !stakeibctypes.IsStAssetDenom(stAssetDenom) {
 		return fmt.Errorf("not a liquid staking token")
 	}
 
 	hostZoneDenom := stakeibctypes.HostZoneDenomFromStAssetDenom(stAssetDenom)
 
-	amount, ok := sdk.NewIntFromString(newData.Amount)
-	if !ok {
+	amount, ok := sdk.NewIntFromString(transferPacketData.Amount)
+	if !ok || amount.IsNegative() {
 		return fmt.Errorf("not a parsable amount field")
 	}
 
-	// Note: newData.denom is ibc denom for st assets - e.g. ibc/xxx
-	var token = sdk.Coin{
-		Denom:  newData.Denom,
-		Amount: amount,
-	}
+	strideAddress := transferPacketData.Receiver
+	redemptionReceiver := autopilotMetadata.IbcReceiver
 
-	if err := token.Validate(); err != nil {
-		return err
-	}
-
-	strideAddress, err := sdk.AccAddressFromBech32(newData.Receiver)
-	if err != nil {
-		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid receiver (%s) in autopilot memo", strideAddress)
-	}
```

Comment:
* Removed token.Validate because `newData.Denom` isn't an actual denom, it's a denom trace path - so this check feels a bit misleading. Plus, if the denom doesn't match our host zone, we error later
* Removed address conversion because downstream we use it as a string instead of address, and it's validated in msg.ValidateBasic

Reply:
> it's validated in msg.ValidateBasic

where does this check happen?

Reply:
[here](https://github.com/Stride-Labs/stride/pull/1030/files#diff-6badb27c023a93b231bdecf85015f77643e5defa71a492747225acaf41d810f5R70), which runs [this](https://github.com/Stride-Labs/stride/blob/main/x/stakeibc/types/message_redeem_stake.go#L47-L50)